Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -1209,115 +1408,70 @@ async def listen(client: SocketModeClient, req: SocketModeRequest):
-                entitlement_reply = entitlement_reply.replace('{user}', f'<@{user_id}>')
-            if '{response}' in entitlement_reply and action_text:
-                entitlement_reply = entitlement_reply.replace('{response}', str(action_text))
-            await send_slack_request_async(client=ASYNC_CLIENT, method='chat.update',
-                                           body={
-                                               'channel': channel,
-                                               'ts': message_ts,
-                                               'text': entitlement_reply,
-                                               'blocks': []
-                                           })
-
-        elif channel and channel[0] == 'D' and ENABLE_DM:
-            # DM
-            await handle_dm(user, text, ASYNC_CLIENT)  # type: ignore
-        else:
-            channel_id = channel
+            await process_entitlement_reply(entitlement_reply, user_id, action_text, channel, message_ts)
 
-            if not integration_context or 'mirrors' not in integration_context:
-                return
-
-            mirrors = json.loads(integration_context['mirrors'])
-            mirror_filter = list(filter(lambda m: m['channel_id'] == channel_id, mirrors))
-            if not mirror_filter:
-                demisto.info("No mirror filter")
-                return
-
-            for mirror in mirror_filter:
-                if mirror['mirror_direction'] == 'FromDemisto' or mirror['mirror_type'] == 'none':
-                    return
-
-                if not mirror['mirrored']:
-                    demisto.info("Not mirrored yet")
-                    # In case the investigation is not mirrored yet
-                    mirror = mirrors.pop(mirrors.index(mirror))
-                    if mirror['mirror_to'] and mirror['mirror_direction'] and mirror['mirror_type']:
-                        investigation_id = mirror['investigation_id']
-                        mirror_type = mirror['mirror_type']
-                        auto_close = mirror['auto_close']
-                        direction = mirror['mirror_direction']
-                        if isinstance(auto_close, str):
-                            auto_close = bool(strtobool(auto_close))
-                        demisto.info(f'Mirroring: {investigation_id}')
-                        demisto.mirrorInvestigation(investigation_id,
-                                                    f'{mirror_type}:{direction}', auto_close)
-                        mirror['mirrored'] = True
-                        mirrors.append(mirror)
-                        set_to_integration_context_with_retries({'mirrors': mirrors},
-                                                                OBJECTS_TO_KEYS, SYNC_CONTEXT)
+        # If a message has made it here, we need to check if the message is being mirrored. If not, we will ignore it.
+        else:
+            await process_mirror(channel, text, user)
```

Comment:
I think before calling `process_mirror` we should check if mirroring is enabled.